lib/grape-swagger.rb

```diff
@@ -49,6 +40,26 @@ def add_swagger_documentation(options = {})
 
       private
 
+      def version_for(options)
+        options[:version] = version if version
+      end
+
+      def combine_routes(app, doc_klass)
```

Comment:
You did lots of good simplification in this file and extracting this block to a method is a huge improvement. It still has an enormous complexity - would love to see this method split down further if possible.